src/GNDStk/Tree/test/read.test.cpp

```diff
@@ -0,0 +1,110 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+using namespace njoy::GNDStk;
+
+SCENARIO("Testing GNDStk tree read() and operator>>") {
+   Tree<> t1, t2, t3, t4;
+
+   GIVEN("Some trees") {
+      WHEN("We read() one from a file") {
+         const std::string filename = "n-008_O_016.xml";
+         t1.read(filename);
+         CHECK(!t1.empty()); // we get something
```

Comment:
Another case of code constructs not existing in a vacuum. Any test of read()'s correctness invariably involves something else, for example a write function in order to compare against a vetted standard result, or a query function in order to pull out data and examine them. While I wrote this a long time ago and don't recall precisely my motive, I suspect that I kept it small, testing only that we read something, because I knew that the real tests would end up being somewhere else (for example, in a test of write(), for which we read() first).